### PR TITLE
kdenlive: Use ffmpeg-full for mlt

### LINKED ISSUE #272526: kdenlive: Unsupported video codec: libaom-av1
### Describe the bug
kdenlive can't encode to AV1 due to missing access to libaom-av1

### Steps To Reproduce
1. Start a new project.
2. Project → Render
3. Presets → Ultra-High Definition (4K)

### Expected behavior

WebM-AV1/Opus (libre) is available. instead it is disabled.

### Screenshots
Screenshot of Render dialog with AV1 disabled.

### Additional context

This is strange because the package appears to try an use the `ffmpeg-full`:

https://github.com/NixOS/nixpkgs/blob/91050ea1e57e50388fa87a3302ba12d188ef723a/pkgs/applications/kde/kdenlive/default.nix#L25

I also have ffmpeg installed in my PATH with libaom-av1 support:

<details>
<summary><code>% ffmpeg -help encoder=libaom-av1</code></summary>

```
ffmpeg version 6.0 Copyright (c) 2000-2023 the FFmpeg developers
  built with gcc 12.3.0 (GCC)
  configuration: --disable-static --prefix=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ffmpeg-full-6.0 --target_os=linux --arch=x86_64 --pkg-config=pkg-config --enable-gpl --enable-version3 --disable-nonfree --enable-shared --enable-pic --disable-small --enable-runtime-cpudetect --enable-gray --enable-swscale-alpha --enable-hardcoded-tables --enable-safe-bitstream-reader --enable-pthreads --disable-w32threads --disable-os2threads --enable-network --enable-pixelutils --datadir=/nix/store/k3696rn7b276i7d8yn53w64h6gn2h0n5-ffmpeg-full-6.0-data/share/ffmpeg --enable-ffmpeg --enable-ffplay --enable-ffprobe --bindir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ffmpeg-full-6.0-bin/bin --enable-avcodec --enable-avdevice --enable-avfilter --enable-avformat --enable-avutil --enable-postproc --enable-swresample --enable-swscale --libdir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ffmpeg-full-6.0-lib/lib --incdir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ffmpeg-full-6.0-dev/include --enable-doc --enable-htmlpages --enable-manpages --mandir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ffmpeg-full-6.0-man/share/man --enable-podpages --enable-txtpages --docdir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ffmpeg-full-6.0-doc/share/doc/ffmpeg --enable-alsa --enable-bzlib --enable-libcelt --enable-cuda --enable-cuda-llvm --enable-libdav1d --disable-libfdk-aac --disable-libflite --enable-fontconfig --enable-libfreetype --enable-frei0r --enable-libfribidi --enable-libgme --enable-gnutls --enable-libgsm --enable-ladspa --enable-libmp3lame --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libdc1394 --enable-libdrm --enable-iconv --enable-libjack --enable-libmfx --enable-libmodplug --enable-libmysofa --enable-libopus --enable-libplacebo --enable-librsvg --enable-libsrt --enable-libssh --disable-libtensorflow --enable-libtheora --enable-libv4l2 --enable-v4l2-m2m --enable-vaapi --enable-vdpau --enable-libvorbis --enable-libvmaf --enable-libvpx --enable-libwebp --enable-xlib --enable-libxcb --enable-libxcb-shm --enable-libxcb-xfixes --enable-libxcb-shape --enable-libxml2 --enable-lzma --enable-cuvid --enable-nvdec --enable-nvenc --enable-openal --enable-opencl --enable-libopencore-amrnb --disable-opengl --enable-libopenh264 --enable-libopenjpeg --enable-libopenmpt --enable-libpulse --enable-librav1e --enable-libsvtav1 --disable-librtmp --enable-sdl2 --enable-libsoxr --enable-libspeex --enable-libvidstab --enable-libvo-amrwbenc --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzmq --enable-libzimg --enable-zlib --enable-vulkan --enable-libglslang --enable-libsmbclient --disable-debug --enable-optimizations --disable-extra-warnings --disable-stripping
  libavutil      58.  2.100 / 58.  2.100
  libavcodec     60.  3.100 / 60.  3.100
  libavformat    60.  3.100 / 60.  3.100
  libavdevice    60.  1.100 / 60.  1.100
  libavfilter     9.  3.100 /  9.  3.100
  libswscale      7.  1.100 /  7.  1.100
  libswresample   4. 10.100 /  4. 10.100
  libpostproc    57.  1.100 / 57.  1.100
Encoder libaom-av1 [libaom AV1]:
    General capabilities: dr1 delay threads 
    Threading capabilities: other
    Supported pixel formats: yuv420p yuv422p yuv444p gbrp yuv420p10le yuv422p10le yuv444p10le yuv420p12le yuv422p12le yuv444p12le gbrp10le gbrp12le gray gray10le gray12le
libaom-av1 encoder AVOptions:
  -cpu-used          <int>        E..V....... Quality/Speed ratio modifier (from 0 to 8) (default 1)
  -auto-alt-ref      <int>        E..V....... Enable use of alternate reference frames (2-pass only) (from -1 to 2) (default -1)
  -lag-in-frames     <int>        E..V....... Number of frames to look ahead at for alternate reference frame selection (from -1 to INT_MAX) (default -1)
  -arnr-max-frames   <int>        E..V....... altref noise reduction max frame count (from -1 to INT_MAX) (default -1)
  -arnr-strength     <int>        E..V....... altref noise reduction filter strength (from -1 to 6) (default -1)
  -aq-mode           <int>        E..V....... adaptive quantization mode (from -1 to 4) (default -1)
     none            0            E..V....... Aq not used
     variance        1            E..V....... Variance based Aq
     complexity      2            E..V....... Complexity based Aq
     cyclic          3            E..V....... Cyclic Refresh Aq
  -error-resilience  <flags>      E..V....... Error resilience configuration (default 0)
     default                      E..V....... Improve resiliency against losses of whole frames
  -crf               <int>        E..V....... Select the quality for constant quality mode (from -1 to 63) (default -1)
  -static-thresh     <int>        E..V....... A change threshold on blocks below which they will be skipped by the encoder (from 0 to INT_MAX) (default 0)
  -drop-threshold    <int>        E..V....... Frame drop threshold (from INT_MIN to INT_MAX) (default 0)
  -denoise-noise-level <int>        E..V....... Amount of noise to be removed (from -1 to INT_MAX) (default -1)
  -denoise-block-size <int>        E..V....... Denoise block size  (from -1 to INT_MAX) (default -1)
  -undershoot-pct    <int>        E..V....... Datarate undershoot (min) target (%) (from -1 to 100) (default -1)
  -overshoot-pct     <int>        E..V....... Datarate overshoot (max) target (%) (from -1 to 1000) (default -1)
  -minsection-pct    <int>        E..V....... GOP min bitrate (% of target) (from -1 to 100) (default -1)
  -maxsection-pct    <int>        E..V....... GOP max bitrate (% of target) (from -1 to 5000) (default -1)
  -frame-parallel    <boolean>    E..V....... Enable frame parallel decodability features (default auto)
  -tiles             <image_size> E..V....... Tile columns x rows
  -tile-columns      <int>        E..V....... Log2 of number of tile columns to use (from -1 to 6) (default -1)
  -tile-rows         <int>        E..V....... Log2 of number of tile rows to use (from -1 to 6) (default -1)
  -row-mt            <boolean>    E..V....... Enable row based multi-threading (default auto)
  -enable-cdef       <boolean>    E..V....... Enable CDEF filtering (default auto)
  -enable-global-motion <boolean>    E..V....... Enable global motion (default auto)
  -enable-intrabc    <boolean>    E..V....... Enable intra block copy prediction mode (default auto)
  -enable-restoration <boolean>    E..V....... Enable Loop Restoration filtering (default auto)
  -usage             <int>        E..V....... Quality and compression efficiency vs speed trade-off (from 0 to INT_MAX) (default good)
     good            0            E..V....... Good quality
     realtime        1            E..V....... Realtime encoding
     allintra        2            E..V....... All Intra encoding
  -tune              <int>        E..V....... The metric that the encoder tunes for. Automatically chosen by the encoder by default (from -1 to 1) (default -1)
     psnr            0            E..V.......
     ssim            1            E..V.......
  -still-picture     <boolean>    E..V....... Encode in single frame mode (typically used for still AVIF images). (default false)
  -enable-rect-partitions <boolean>    E..V....... Enable rectangular partitions (default auto)
  -enable-1to4-partitions <boolean>    E..V....... Enable 1:4/4:1 partitions (default auto)
  -enable-ab-partitions <boolean>    E..V....... Enable ab shape partitions (default auto)
  -enable-angle-delta <boolean>    E..V....... Enable angle delta intra prediction (default auto)
  -enable-cfl-intra  <boolean>    E..V....... Enable chroma predicted from luma intra prediction (default auto)
  -enable-filter-intra <boolean>    E..V....... Enable filter intra predictor (default auto)
  -enable-intra-edge-filter <boolean>    E..V....... Enable intra edge filter (default auto)
  -enable-smooth-intra <boolean>    E..V....... Enable smooth intra prediction mode (default auto)
  -enable-paeth-intra <boolean>    E..V....... Enable paeth predictor in intra prediction (default auto)
  -enable-palette    <boolean>    E..V....... Enable palette prediction mode (default auto)
  -enable-flip-idtx  <boolean>    E..V....... Enable extended transform type (default auto)
  -enable-tx64       <boolean>    E..V....... Enable 64-pt transform (default auto)
  -reduced-tx-type-set <boolean>    E..V....... Use reduced set of transform types (default auto)
  -use-intra-dct-only <boolean>    E..V....... Use DCT only for INTRA modes (default auto)
  -use-inter-dct-only <boolean>    E..V....... Use DCT only for INTER modes (default auto)
  -use-intra-default-tx-only <boolean>    E..V....... Use default-transform only for INTRA modes (default auto)
  -enable-ref-frame-mvs <boolean>    E..V....... Enable temporal mv prediction (default auto)
  -enable-reduced-reference-set <boolean>    E..V....... Use reduced set of single and compound references (default auto)
  -enable-obmc       <boolean>    E..V....... Enable obmc (default auto)
  -enable-dual-filter <boolean>    E..V....... Enable dual filter (default auto)
  -enable-diff-wtd-comp <boolean>    E..V....... Enable difference-weighted compound (default auto)
  -enable-dist-wtd-comp <boolean>    E..V....... Enable distance-weighted compound (default auto)
  -enable-onesided-comp <boolean>    E..V....... Enable one sided compound (default auto)
  -enable-interinter-wedge <boolean>    E..V....... Enable interinter wedge compound (default auto)
  -enable-interintra-wedge <boolean>    E..V....... Enable interintra wedge compound (default auto)
  -enable-masked-comp <boolean>    E..V....... Enable masked compound (default auto)
  -enable-interintra-comp <boolean>    E..V....... Enable interintra compound (default auto)
  -enable-smooth-interintra <boolean>    E..V....... Enable smooth interintra mode (default auto)
  -aom-params        <dictionary> E..V....... Set libaom options using a :-separated list of key=value pairs
```

</details>

### Notify maintainers

@turion 

### Metadata

The exact package I am using is `/nix/store/qfd2d8c3s42j83ffhlq8wpg4y5hq95w2-kdenlive-23.08.3` which is available in https://cache.nixos.org.

`% nix-shell -p nix-info --run "nix-info -m"`
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.64, NixOS, 24.05 (Uakari), 24.05pre555097.91050ea1e57e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### PR DESCRIPTION
**Pull request by turion**
## Description of changes

Fixes https://github.com/NixOS/nixpkgs/issues/272526

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---
@kevincox can you test whether this fixes https://github.com/NixOS/nixpkgs/issues/272526?

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by kevincox — APPROVED** [review]
I can confirm that this makes the option available and rendering to AV1 works.

File: pkgs/applications/kde/kdenlive/default.nix [hunk 1]
```diff
@@ -36,6 +36,11 @@
 , wrapGAppsHook
 }:
 
+let
+  mlt-full = mlt.override {
```
[the remark this answers is not in the document]
Comment (turion) [reply pkgs/applications/kde/kdenlive/default.nix:40]: Hmm, yes, we could move the override to `all-packages.nix`.

File: pkgs/applications/kde/kdenlive/default.nix [hunk 2]
```diff
@@ -36,6 +36,11 @@
...
+    ffmpeg = ffmpeg-full;
```
[the remark this answers is not in the document]
Comment (kevincox) [reply pkgs/applications/kde/kdenlive/default.nix:41]: Yeah, makes sense to improve at least this package for now while we discuss making this the default.

File: pkgs/applications/kde/kdenlive/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevincox) [reply pkgs/applications/kde/kdenlive/default.nix:40]: Yeah. That is what I would have done. But I don't think there are any strong guidelines here. I wouldn't block either way if you prefer this.

**Comment by turion** [comment]
@Mindavi I know you mean well, but I wasn't done with this PR yet :sweat_smile: Two conversations weren't resolved.

**Comment by Mindavi** [comment]
Oops! I missed the first conversation, figured the second one could be resolved later

**Pull request merged**